publication-model/src/main/java/no/unit/nva/api/PublicationResponse.java

```diff
@@ -81,20 +85,30 @@ public static PublicationResponse fromPublication(Publication publication) {
         response.setHandle(publication.getHandle());
         response.setLink(publication.getLink());
         response.setEntityDescription(publication.getEntityDescription());
-        response.setAssociatedArtifacts(publication.getAssociatedArtifacts());
         response.setDoi(publication.getDoi());
         response.setProjects(publication.getProjects());
         response.setFundings(publication.getFundings());
         response.setSubjects(publication.getSubjects());
         response.setContext(PublicationContext.getContext(publication));
-        response.setAssociatedArtifacts(publication.getAssociatedArtifacts());
+        response.setAssociatedArtifacts(getAllNonInternalFiles(publication));
         response.setAdditionalIdentifiers(publication.getAdditionalIdentifiers());
         response.setRightsHolder(publication.getRightsHolder());
         response.setAllowedOperations(Set.of());
         response.setImportDetails(publication.getImportDetails());
         return response;
     }
 
+    private static AssociatedArtifactList getAllNonInternalFiles(Publication publication) {
+        var artifacts = publication.getAssociatedArtifacts().stream()
+                            .filter(PublicationResponse::isNotInternalFile)
+                            .toList();
+        return new AssociatedArtifactList(artifacts);
+    }
+
+    private static boolean isNotInternalFile(AssociatedArtifact artifact) {
+        return !(artifact instanceof PendingInternalFile) && !(artifact instanceof InternalFile);
```

Comment:
when thinking about if, should we hide rejected also? An PendingInternalFile can become Rejected

Reply:
should maybe use isVisibleForNonOwner also here so that embargo is not exposed?

Reply:
Fikk denne fra Jan Erik:
```
Ja, gi dei med skrivetilgang tilgang til interne filer.
Nei, filer som er “i saksganf/flyt” vises ikkje fram i API

NB! Det siste er eit forsøk på å redusere kompleksitet, om det ikkje bidrar til det, så kan vi endre på det!
Eg oppfatter ikkje at vi har nokon klare brukerønsker knyttet til API, som omhandler filer som ikkje er i ein “sluttstatus”.
```

Virker da som at kun OpenFile skal være "offentlig"

Reply:
Enig @torbjokv ?